src/server/bitops_family.cc

```diff
@@ -57,14 +65,88 @@ constexpr int32_t GetByteIndex(uint32_t offset) noexcept {
   return offset / OFFSET_FACTOR;
 }
 
-uint8_t GetByteValue(const std::string& str, uint32_t offset) {
+uint8_t GetByteValue(std::string_view str, uint32_t offset) {
   return static_cast<uint8_t>(str[GetByteIndex(offset)]);
 }
 
 constexpr bool CheckBitStatus(uint8_t byte, uint32_t offset) {
   return byte & (0x1 << offset);
 }
 
+constexpr std::uint8_t CountBitsRange(std::uint8_t byte, std::uint8_t from, uint8_t to) {
+  int count = 0;
+  for (int i = from; i < to; i++) {
+    count += CheckBitStatus(byte, GetNormalizedBitIndex(i));
+  }
+  return count;
+}
+
+constexpr std::uint8_t BitOnCount(std::uint8_t byte) {
+  return NIBBLE_LOOKUP[byte & 0x0F] + NIBBLE_LOOKUP[byte >> 4];
+}
+
+std::uint32_t CountSetAllBytes(std::string_view at, int32_t start, int32_t end) {
+  if (start >= end || start < 0 || end < 0) {
```

Comment:
you already check similar conditions in the external function. Are you sure start and end can be negative or 
`start > end` ?

Reply:
if it can not happen, please use unsigned types for internal functions to reduce number of edge cases

Reply:
The less than 0 is removed, for the other case, this can be called for 2 places, so just safe guard, otherwise I need to call it from the "outside" again.